lib/portlayer/exec/container.go

```diff
@@ -234,6 +234,17 @@ func (c *Container) updateState(s State) State {
 	return prevState
 }
 
+// transitionState changes the container state to finalState if the current state is initialState.
+func (c *Container) transitionState(initialState, finalState State) {
+	c.m.Lock()
+	defer c.m.Unlock()
+
+	if c.state == initialState {
+		c.state = finalState
+		log.Debugf("Set container %s state: %s", c.ExecConfig.ID, finalState)
+	}
```

Comment:
Also I think this function should return an error to the caller so that caller can know whether transition succeeded or not.